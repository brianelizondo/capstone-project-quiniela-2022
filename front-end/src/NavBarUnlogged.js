import React from 'react';
import { NavDropdown } from 'react-bootstrap';

// options to show when the user is unlogged
function NavBarUnlogged() {
    return (
        <>
        <NavDropdown.Item href="/login">Login</NavDropdown.Item>
        <NavDropdown.Item href="/register">Register</NavDropdown.Item>
        </>
    );
}

export default NavBarUnlogged;