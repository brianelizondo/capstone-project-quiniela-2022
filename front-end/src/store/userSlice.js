import { createSlice } from '@reduxjs/toolkit';

export const userSlice = createSlice({
    name: 'user',
    initialState: {
        id: null, 
        firstName: null, 
        lastName: null, 
        email: null, 
        username: null, 
        isAdmin: false,
        token: null,
        isAuthenticated: false
    },
    reducers: {
        login: (state, action) => {
            state.id = action.payload.user.id;
            state.firstName = action.payload.user.firstName;
            state.lastName = action.payload.user.lastName;
            state.email = action.payload.user.email;
            state.username = action.payload.user.username;
            state.isAdmin = action.payload.user.isAdmin;
            state.token = action.payload.token;
            state.isAuthenticated = true;
        },
        update: (state, action) => {
            state = action.payload
        },
        logout: (state) => {
            state.id = null;
            state.firstName = null;
            state.lastName = null;
            state.email = null;
            state.username = null;
            state.isAdmin = null;
            state.token = null;
            state.isAuthenticated = false;
        }
    }
})
  
// Action creators are generated for each case reducer function
export const { login, update, logout } = userSlice.actions;
export default userSlice.reducer;