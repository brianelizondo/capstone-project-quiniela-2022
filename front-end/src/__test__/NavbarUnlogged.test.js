import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { persistStore } from 'redux-persist';
import store from '../store/store';
let persistor = persistStore(store);

import NavBarUnlogged from '../NavBarUnlogged';

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'NavBarUnlogged' component without crashing", () => {
        render(
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistor}>
                    <MemoryRouter>
                        <NavBarUnlogged />
                    </MemoryRouter>
                </PersistGate>
            </Provider>
        );
        const textFound = screen.getByText("Login");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'NavBarUnlogged' matches snapshot", () => {
        const {asFragment} = render(
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistor}>
                    <MemoryRouter>
                        <NavBarUnlogged />
                    </MemoryRouter>
                </PersistGate>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
