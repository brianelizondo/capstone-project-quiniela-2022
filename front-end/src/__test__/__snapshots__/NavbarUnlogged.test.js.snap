// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'NavBarUnlogged' matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="dropdown-item"
    data-rr-ui-dropdown-item=""
    href="/login"
  >
    Login
  </a>
  <a
    class="dropdown-item"
    data-rr-ui-dropdown-item=""
    href="/register"
  >
    Register
  </a>
</DocumentFragment>
`;
